clients/java/zpe/src/main/java/com/yahoo/athenz/zpe/AuthZpeClient.java

```diff
@@ -386,6 +386,24 @@ public static AccessCheckStatus allowAccess(String roleToken, String angResource
             if (rToken.validate(getZtsPublicKey(rToken.getKeyId()), allowedOffset, false, null) == false) {
                 LOG.error("allowAccess: Authorization denied. Authentication of token failed for token="
                         + rToken.getSignedToken());
+
+                // check the token expiration
+                long now    = System.currentTimeMillis() / 1000;
+                long expiry = rToken.getExpiryTime();
+                if (expiry != 0 && expiry < now) {
+                    String signedToken = rToken.getSignedToken();
+                    LOG.error("allowAccess: Authorization denied. Token expired. now=" +
```

Comment:
Let's move the LOG.error from line 387 after this expiry block check. Otherwise, we'll be logging the failure twice.